File: sdks/local-rp/dart/test/conformance/tickets_test.dart
import 'package:linkkeys_local_rp/src/crypto/crypto.dart';
import 'package:test/test.dart';

import '../testutil/fixtures.dart';

void main() {
  group('tickets.json conformance', () {
    test('hash pairs match fingerprint routine', () async {
      final d = loadJson('tickets.json') as Map<String, dynamic>;
      final cases = (d['cases'] as List).cast<Map<String, dynamic>>();
      expect(cases, isNotEmpty);
      for (final c in cases) {
        final ticket = hex(c['ticket_hex'] as String);
        expect(ticket.length, equals(32));
        expect(await Crypto.fingerprint(ticket), equals(c['sha256_hex']));
      }
    });
  });
}

File: sdks/local-rp/php/tests/conformance/TicketsTest.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/../bootstrap.php';

use LinkKeys\LocalRp\Crypto;

$vectors = loadJson(__DIR__ . '/../../../conformance/tickets.json');

foreach ($vectors['cases'] as $case) {
    TestKit::test('tickets.' . $case['name'], function () use ($case) {
        $ticket = hexToBytes($case['ticket_hex']);
        TestKit::assertEquals($case['sha256_hex'], Crypto::fingerprint($ticket));
    });
}

exit(TestKit::summary('TicketsTest') === 0 ? 0 : 1);
